Add number of blocked bugs ad a feature

rhcu opened the pull request:
Fixes #108

marco-c commented:
Could you check if it makes any difference in the `bug` or in the `tracking` models? (Run them before/after adding the feature, and post here the output before/after)

rhcu commented:
Before adding the feature to the `bug` model:
Before adding the feature:
```
assiyakhuzyakhmetova$ python -i run.py —train —goal bug
X: (6651, 241662), y: (6651,)
X_train: (1984, 241662), y_train: (1984,)
X_test: (666, 241662), y_test: (666,)
CV Accuracy: f0.9304553068372163 (+/- 0.020415517806888713)
Feature ranking (top 20 features):
1. 'comments__regression' (0.07854502648115158)
2. 'first_comment__str' (0.049220938235521317)
3. 'first_comment__mozilla' (0.04819164052605629)
4. 'data__severity=critical' (0.044781479984521866)
5. 'comments__nightly' (0.040189243853092194)
6. 'comments__reproduce' (0.032989416271448135)
7. 'first_comment__10' (0.03132477402687073)
8. 'first_comment__expected' (0.029585424810647964)
9. 'comments__issue' (0.02903810515999794)
10. 'comments__fix' (0.028362935408949852)
11. 'comments__pushlog' (0.027619460597634315)
12. 'comments__remove' (0.025549449026584625)
13. 'comments__marked' (0.025051292032003403)
14. 'first_comment__attachment' (0.023658128455281258)
15. 'first_comment__regression' (0.022702306509017944)
16. 'comments__support' (0.018896792083978653)
17. 'first_comment____code_reference_url__' (0.018880564719438553)
18. 'comments__use' (0.018595391884446144)
19. 'comments__57' (0.018031595274806023)
20. 'comments__work' (0.014156877994537354)
No confidence threshold - 666 classified
[[499 50]
[ 10 107]]
pre rec spe f1 geo iba sup

1 0.98 0.91 0.91 0.94 0.91 0.83 549
0 0.68 0.91 0.91 0.78 0.91 0.83 117

avg / total 0.93 0.91 0.91 0.91 0.91 0.83 666

Confidence threshold > 0.6 - 625 classified
[[479 39]
[ 5 102]]
pre rec spe f1 geo iba sup

1 0.99 0.92 0.95 0.96 0.94 0.88 518
0 0.72 0.95 0.92 0.82 0.94 0.88 107

avg / total 0.94 0.93 0.95 0.93 0.94 0.88 625

Confidence threshold > 0.7 - 602 classified
[[471 27]
[ 3 101]]
pre rec spe f1 geo iba sup

1 0.99 0.95 0.97 0.97 0.96 0.92 498
0 0.79 0.97 0.95 0.87 0.96 0.92 104

avg / total 0.96 0.95 0.97 0.95 0.96 0.92 602

Confidence threshold > 0.8 - 570 classified
[[452 21]
[ 2 95]]
pre rec spe f1 geo iba sup

1 1.00 0.96 0.98 0.98 0.97 0.93 473
0 0.82 0.98 0.96 0.89 0.97 0.94 97

avg / total 0.97 0.96 0.98 0.96 0.97 0.93 570

Confidence threshold > 0.9 - 503 classified
[[413 9]
[ 1 80]]
pre rec spe f1 geo iba sup

1 1.00 0.98 0.99 0.99 0.98 0.97 422
0 0.90 0.99 0.98 0.94 0.98 0.97 81

avg / total 0.98 0.98 0.99 0.98 0.98 0.97 503
```
After adding the feature to the `bug` model:
```
(env) assiyakhuzyakhmetova$ python -i run.py --train --goal bug
X: (6651, 241663), y: (6651,)
X_train: (1984, 241663), y_train: (1984,)
X_test: (666, 241663), y_test: (666,)
CV Accuracy: f0.9309628952845033 (+/- 0.02527295877105665)
Feature ranking (top 20 features):
1. 'comments__regression' (0.07854502648115158)
2. 'first_comment__str' (0.049220938235521317)
3. 'first_comment__mozilla' (0.04819164052605629)
4. 'data__severity=critical' (0.044781479984521866)
5. 'comments__nightly' (0.040189243853092194)
6. 'comments__reproduce' (0.032989416271448135)
7. 'first_comment__10' (0.03132477402687073)
8. 'first_comment__expected' (0.029585424810647964)
9. 'comments__issue' (0.02903810515999794)
10. 'comments__fix' (0.028362935408949852)
11. 'comments__pushlog' (0.027619460597634315)
12. 'comments__remove' (0.025549449026584625)
13. 'comments__marked' (0.025051292032003403)
14. 'first_comment__attachment' (0.023658128455281258)
15. 'first_comment__regression' (0.022702306509017944)
16. 'comments__support' (0.018896792083978653)
17. 'first_comment____code_reference_url__' (0.018880564719438553)
18. 'comments__use' (0.018595391884446144)
19. 'comments__57' (0.018031595274806023)
20. 'comments__work' (0.014156877994537354)
No confidence threshold - 666 classified
[[499  50]
 [ 10 107]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.98      0.91      0.91      0.94      0.91      0.83       549
          0       0.68      0.91      0.91      0.78      0.91      0.83       117

avg / total       0.93      0.91      0.91      0.91      0.91      0.83       666


Confidence threshold > 0.6 - 625 classified
[[479  39]
 [  5 102]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.99      0.92      0.95      0.96      0.94      0.88       518
          0       0.72      0.95      0.92      0.82      0.94      0.88       107

avg / total       0.94      0.93      0.95      0.93      0.94      0.88       625


Confidence threshold > 0.7 - 602 classified
[[471  27]
 [  3 101]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.99      0.95      0.97      0.97      0.96      0.92       498
          0       0.79      0.97      0.95      0.87      0.96      0.92       104

avg / total       0.96      0.95      0.97      0.95      0.96      0.92       602


Confidence threshold > 0.8 - 570 classified
[[452  21]
 [  2  95]]
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.96      0.98      0.98      0.97      0.93       473
          0       0.82      0.98      0.96      0.89      0.97      0.94        97

avg / total       0.97      0.96      0.98      0.96      0.97      0.93       570


Confidence threshold > 0.9 - 503 classified
[[413   9]
 [  1  80]]
                   pre       rec       spe        f1       geo       iba       sup

          1       1.00      0.98      0.99      0.99      0.98      0.97       422
          0       0.90      0.99      0.98      0.94      0.98      0.97        81

avg / total       0.98      0.98      0.99      0.98      0.98      0.97       503
```

So, CV accuracy of `bug` model **increased by ~0.0002**.

`Tracking` model without feature:

```(env) Assiyas-MacBook-Pro:bugbug assiyakhuzyakhmetova$ python -i run.py --train --goal tracking
Current value for field cf_status_firefox60:
---
is different from previous value:
fixed
Current value for field cf_status_firefox58:
---
is different from previous value:
unaffected
X: (59126, 726468), y: (59126,)
X_train: (3274, 726468), y_train: (3274,)
X_test: (5913, 726468), y_test: (5913,)
CV Accuracy: f0.8671197881703587 (+/- 0.03183299484999692)
Feature ranking (top 20 features):
1. 'comments__tracking' (0.13687747716903687)
2. 'comments__release' (0.0890636220574379)
3. 'data__landings' (0.07378990948200226)
4. 'data__keywords-regression=True' (0.06697406619787216)
5. 'comments__regression' (0.04812086746096611)
6. 'comments__com' (0.04181256145238876)
7. 'comments__beta' (0.037207599729299545)
8. 'comments__mozilla' (0.031325168907642365)
9. 'data__patches' (0.03012906014919281)
10. 'comments__breakdown' (0.027038447558879852)
11. 'comments__duplicate' (0.025633450597524643)
12. 'data__keywords-sec-=True' (0.025562042370438576)
13. 'comments__uplift' (0.023842405527830124)
14. 'comments__requested' (0.019436821341514587)
15. 'title__beta' (0.013555618934333324)
16. 'comments____code_reference_url__' (0.01275682169944048)
17. 'comments__nightly' (0.012427618727087975)
18. 'title__2018' (0.011921490542590618)
19. 'comments__platform' (0.011142649687826633)
20. 'title____file_reference__' (0.010730918496847153)
No confidence threshold - 5913 classified
[[ 173   36]
 [ 629 5075]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.22      0.83      0.89      0.34      0.86      0.73       209
          0       0.99      0.89      0.83      0.94      0.86      0.74      5704

avg / total       0.97      0.89      0.83      0.92      0.86      0.74      5913


Confidence threshold > 0.6 - 5442 classified
[[ 162   30]
 [ 440 4810]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.27      0.84      0.92      0.41      0.88      0.77       192
          0       0.99      0.92      0.84      0.95      0.88      0.78      5250

avg / total       0.97      0.91      0.85      0.93      0.88      0.78      5442


Confidence threshold > 0.7 - 4809 classified
[[ 149   23]
 [ 309 4328]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.33      0.87      0.93      0.47      0.90      0.80       172
          0       0.99      0.93      0.87      0.96      0.90      0.81      4637

avg / total       0.97      0.93      0.87      0.95      0.90      0.81      4809


Confidence threshold > 0.8 - 3650 classified
[[ 134    7]
 [ 205 3304]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.40      0.95      0.94      0.56      0.95      0.90       141
          0       1.00      0.94      0.95      0.97      0.95      0.89      3509

avg / total       0.97      0.94      0.95      0.95      0.95      0.89      3650


Confidence threshold > 0.9 - 2316 classified
[[ 106    2]
 [  93 2115]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.53      0.98      0.96      0.69      0.97      0.94       108
          0       1.00      0.96      0.98      0.98      0.97      0.94      2208

avg / total       0.98      0.96      0.98      0.96      0.97      0.94      2316
```

`Tracking` model with the feature:
```
assiyakhuzyakhmetova$ python -i run.py --train --goal tracking
Current value for field cf_status_firefox60:
---
is different from previous value:
fixed
Current value for field cf_status_firefox58:
---
is different from previous value:
unaffected
X: (59126, 726469), y: (59126,)
X_train: (3274, 726469), y_train: (3274,)
X_test: (5913, 726469), y_test: (5913,)
CV Accuracy: f0.8671197881703587 (+/- 0.03832946363889364)
Feature ranking (top 20 features):
1. 'comments__tracking' (0.13842380046844482)
2. 'comments__release' (0.09126247465610504)
3. 'data__landings' (0.06850361824035645)
4. 'data__keywords-regression=True' (0.067457415163517)
5. 'comments__regression' (0.04888955131173134)
6. 'comments__com' (0.042230479419231415)
7. 'comments__beta' (0.038692791014909744)
8. 'comments__mozilla' (0.029203079640865326)
9. 'comments__uplift' (0.02612215280532837)
10. 'data__keywords-sec-=True' (0.025613849982619286)
11. 'comments__breakdown' (0.025458820164203644)
12. 'comments__duplicate' (0.02536763995885849)
13. 'data__patches' (0.025213416665792465)
14. 'comments__requested' (0.01624530926346779)
15. 'title__beta' (0.013902162201702595)
16. 'title____file_reference__' (0.013424703851342201)
17. 'title__2018' (0.011893589049577713)
18. 'comments__platform' (0.011380085721611977)
19. 'comments__approval' (0.011237893253564835)
20. 'comments____code_reference_url__' (0.010764162987470627)
No confidence threshold - 5913 classified
[[ 171   38]
 [ 638 5066]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.21      0.82      0.89      0.34      0.85      0.72       209
          0       0.99      0.89      0.82      0.94      0.85      0.73      5704

avg / total       0.96      0.89      0.82      0.92      0.85      0.73      5913


Confidence threshold > 0.6 - 5443 classified
[[ 164   32]
 [ 445 4802]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.27      0.84      0.92      0.41      0.88      0.76       196
          0       0.99      0.92      0.84      0.95      0.88      0.77      5247

avg / total       0.97      0.91      0.84      0.93      0.88      0.77      5443


Confidence threshold > 0.7 - 4748 classified
[[ 149   17]
 [ 312 4270]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.32      0.90      0.93      0.48      0.91      0.83       166
          0       1.00      0.93      0.90      0.96      0.91      0.84      4582

avg / total       0.97      0.93      0.90      0.95      0.91      0.84      4748


Confidence threshold > 0.8 - 3674 classified
[[ 133    7]
 [ 215 3319]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.38      0.95      0.94      0.55      0.94      0.89       140
          0       1.00      0.94      0.95      0.97      0.94      0.89      3534

avg / total       0.97      0.94      0.95      0.95      0.94      0.89      3674


Confidence threshold > 0.9 - 2284 classified
[[ 105    1]
 [  95 2083]]
                   pre       rec       spe        f1       geo       iba       sup

          1       0.53      0.99      0.96      0.69      0.97      0.95       106
          0       1.00      0.96      0.99      0.98      0.97      0.94      2178

avg / total       0.98      0.96      0.99      0.96      0.97      0.94      2284
```
The uncertainty of CV accuracy increased but the value remained the same.
So, seems like it works better with `bug` model, and I will add the feature to it.

marco-c commented:
Could you add a test too in test_bug_features.py?
